### PR TITLE
Add flake8 as a python linter and styleguide

### PR DESCRIPTION
**Pull request by Kapin**
Enforces that the code follows the PEP-8 style guide for python and some linting.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[flake8]
+max-line-length = 120
```
Comment (Trinovantes) [inline setup.cfg]: I have a feeling we might regret this later

File: src/sanaprotocolbuilder/urls.py [hunk 2]
```diff
@@ -6,7 +6,9 @@
 from webapp import views as webappViews
 from rest_framework.authtoken import views as restViews
 
-urlpatterns = patterns('',
+urlpatterns = patterns(
+    '',
```
Comment (Trinovantes) [inline src/sanaprotocolbuilder/urls.py]: I think we can delete this empty clause

File: setup.cfg (hunk 1 above)
Comment (Kapin) [inline setup.cfg]: Why? 120 is standard. The default is 79 which is way too short.

Comment (Trinovantes) [inline setup.cfg]: We can keep it at 120 for now but I have a feeling we may have to increase this later when we start doing weird chaining

Comment (Kapin) [inline setup.cfg]: I'd like to avoid weird chaining...

Comment (Kapin) [inline setup.cfg]: Also you can override any of these by putting a comment on that line with `noqa`

File: src/sanaprotocolbuilder/urls.py (hunk 2 above)
Comment (Kapin) [inline src/sanaprotocolbuilder/urls.py]: will do in a separate PR

Comment (Trinovantes) [inline src/sanaprotocolbuilder/urls.py]: Nevermind, that's supposed to be there. It's for prefixing the views (https://docs.djangoproject.com/en/1.7/topics/http/urls/#the-view-prefix)

**Comment by Trinovantes** [comment]
+1

**Comment by Kapin** [comment]
Will merge after #57

**Comment by Kapin** [comment]
Please do a thorough review of this, I don't things to break.

**Comment by Trinovantes** [comment]
You missed word there :laughing:

File: fabfile.py [hunk 3]
```diff
@@ -1,17 +1,29 @@
...
-env.virtualenv      = 'source /usr/local/bin/virtualenvwrapper.sh'
-env.project_root    = '/opt/sana.protocol_builder'
+env.hosts = ['sanaprotocolbuilder.me']
+env.user = 'root'
+env.virtualenv = 'source /usr/local/bin/virtualenvwrapper.sh'
+env.project_root = '/opt/sana.protocol_builder'
+
 
 def test():
     local('python src/manage.py syncdb --noinput')
```
Comment (Trinovantes) [inline fabfile.py]: Let's add some coloured print statements here too to make it clearer what it's doing

File: src/editor/tests/test_model_procedure.py [hunk 4]
```diff
@@ -2,12 +2,21 @@
...
+        test_user = User.objects.create_user(
+            'TestUser',
+            'test@sanaprotocolbuilder.me',
+            'testpassword'
+        )
         test_user.save()
 
-        Procedure.objects.create(author='tester', title='test procedure', owner=test_user)
+        Procedure.objects.create(
+            author='tester',
```
Comment (Trinovantes) [inline src/editor/tests/test_model_procedure.py]: Align these `=` please. Get the alignment Sublime plugin (http://wbond.net/sublime_packages/alignment)

File: src/homepage/admin.py [hunk 5]
```diff
@@ -1,3 +1,3 @@
-from django.contrib import admin
+# from django.contrib import admin
```
Comment (Trinovantes) [inline src/homepage/admin.py]: This file should be removed later.

File: src/homepage/tests/test_user.py [hunk 6]
```diff
@@ -3,13 +3,19 @@
...
+            'TestUser',
+            'test@sanaprotocolbuilder.me',
+            'testpassword'
+        )
         self.test_user.save()
 
     def test_user_has_token(self):
         try:
             token = Token.objects.get(user=self.test_user)
+            self.assertFalse(token is None)
```
Comment (Trinovantes) [inline src/homepage/tests/test_user.py]: I think it's easier to read if it's `self.assertTrue(token is not None)` to avoid double negatives

File: src/homepage/views.py [hunk 7]
```diff
@@ -1,23 +1,34 @@
...
 def signup(request):
     template = loader.get_template('webapp/signup.html')
     context = RequestContext(request, {
-        'title' : "Sign Up",
+        'title': "Sign Up",
+    })
+    return HttpResponse(template.render(context))
+
+
+def editor(request):
```
Comment (Trinovantes) [inline src/homepage/views.py]: This is removed from `homepage` app

File: src/sanaprotocolbuilder/urls.py (hunk 2 above)
Comment (Trinovantes) [inline src/sanaprotocolbuilder/urls.py]: Let's keep the prefixes on the same line as `patterns(` as a Django convention (to keep it consistent with the docs)

File: fabfile.py (hunk 3 above)
Comment (Kapin) [inline fabfile.py]: separate pr

File: src/homepage/tests/test_user.py (hunk 6 above)
Comment (Kapin) [inline src/homepage/tests/test_user.py]: Fair, can change

File: src/homepage/views.py (hunk 7 above)
Comment (Kapin) [inline src/homepage/views.py]: fixed

File: src/editor/tests/test_model_procedure.py (hunk 4 above)
Comment (Kapin) [inline src/editor/tests/test_model_procedure.py]: Following the python style guide. For named parameters, you align the name, not the =

File: src/sanaprotocolbuilder/urls.py (hunk 2 above)
Comment (Kapin) [inline src/sanaprotocolbuilder/urls.py]: will change

File: src/editor/tests/test_model_procedure.py (hunk 4 above)
Comment (Trinovantes) [inline src/editor/tests/test_model_procedure.py]: :(

**Comment by Trinovantes** [comment]
+1

**Comment by ConnorCimowsky** [comment]
yes-approval-lgtm
